compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1059,7 +1059,18 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
     val (stats1, exprCtx) = withoutMode(Mode.Pattern) {
       typedBlockStats(tree.stats)
     }
-    val expr1 = typedExpr(tree.expr, pt.dropIfProto)(using exprCtx)
+    var expr1 = typedExpr(tree.expr, pt.dropIfProto)(using exprCtx)
+
+    // If unsafe nulls is enabled inside a block but not enabled outside
```

Comment:
The comment does not correspond to the code. `exprCtx` is the context inside the block valid for the final expression. `ctx` is the context outside the block. The cast is performed if unsafeNulls is _disabled_ inside the block but _enabled_ around it. Is this what's intended? Either the comment or the code has to be fixed.

Reply:
I think the comment is correct: `!exprCtx.mode.is(Mode.SafeNulls)` indicates `SafeNulls` is disabled inside the block, which means unsafeNulls is enabled here.

Reply:
The confusion is that the mode is `SafeNulls` which is a negation of `unsafeNulls`. I think the comment correctly corresponds to the code.

Reply:
Yes, my bad. I misread the code.